### PR TITLE
lightning: fix uuid panic when expression index enabled

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/44497

Problem Summary:

### What is changed and how it works?
Add `e.SessionCtx.Vars.TxnCtx` when getting default value for column with default expression.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
manually load data with uuid and it works.

Side effects

- None

Documentation

- None

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
lightning: fix the problem that lightning crashes when target table has a column whose default value is uuid
```

### CODE REVIEW

File: br/tests/lightning_generated_columns/data/gencol.uuid.0 copy.sql
Comment (lance6716) [inline br/tests/lightning_generated_columns/data/gencol.uuid.0 copy.sql:1]: strange file name

File: br/tests/lightning_generated_columns/run.sh [hunk 1]
```diff
@@ -92,4 +92,8 @@ for BACKEND in 'local' 'tidb'; do
     check_contains 'id: 2'
     check_contains 'a: ABC'
     check_contains 'b: CDSFDS'
+
+    run_sql 'ADMIN CHECK TABLE gencol.uuid'
+    run_sql 'SELECT /*+ use_index(gencol.uuid, c) */ COUNT(b) FROM gencol.uuid'
```
Comment (lance6716) [inline br/tests/lightning_generated_columns/run.sh:97]: maybe also check the uuid should not be same by select count distinct

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Review by D3Hunter — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**